### ISSUE TITLE
hxcpp run failed :was killed with test example.zip

### ISSUE THREAD

**Issue by sonygod** [HaxeFoundation/hxcpp#853]
[test.zip](https://github.com/HaxeFoundation/hxcpp/files/3655453/test.zip)

run in hxcpp got nothing error information but crash.

but Hashlink target run well.

**Comment by hughsando** [comment]
To access more than 2Gig of memory, you should use the following defines:
``` -D HXCPP_M64 -D HXCPP_GC_BIG_BLOCKS```

**Comment by sonygod** [comment]
@hughsando  still crash. after add  

-D HXCPP_M64 -D HXCPP_GC_BIG_BLOCKS

my pc is more than 16 g  memory

**Comment by RealyUniqueName** [comment]
Add `-debug` flag to get exception details on crash. It may not help if it's caused by GC.

**Comment by sonygod** [comment]
it seems more than 2 Gig memory  in hxcpp, but less than 2G in hashlink

**Comment by hughsando** [comment]
Yes, hxcpp is certainly using quite a bit of Ram.  There are lots of allocations and stores going on, so I'm not really sure what it is supposed to be.  One possibility is that the memory is getting fragmented. Another is that some of the structures (eg, arrays, anon-object) are bigger in hxcpp due to extra reserved space.
But it is working pretty solidly with:
[build.hxml.zip](https://github.com/HaxeFoundation/hxcpp/files/3657126/build.hxml.zip)

**Comment by sonygod** [comment]
oh, I see,the memory  cost more than 4 Gb,

Is there any plan to improve Ram?

**Comment by hughsando** [comment]
I have made some changes to allow the system to work in low RAM situations, which allows you to use HXCPP_GC_MOVING instead of HXCPP_GC_BIG_BLOCKS.  This keeps the RAM around 2.2 Gig which lets you do some things like replace the typedefs with classes.  Still a lot more than HL I think - not sure where the difference is.  Without knowing specifically how many of each object type is allocated I'm not sure I can o much more.
